### PR TITLE
Quickstart match strategy, triggered rule / template / query rewrite and style changes

### PR DESCRIPTION
**Pull request by JerryCB**
This version of the quickstart incorporates changes made for Zoro Tools:
1. Enable users to specify a match strategy per results column
2. Display any triggered rule, template or query rewrites
3. Extensive style changes implemented by Olya Sanakoev

### CODE REVIEW

File: src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java [hunk 1]
```diff
@@ -264,7 +291,11 @@ protected ModelAndView handleSearch(HttpServletRequest request, HttpServletRespo
                     model.put("time" + i, System.currentTimeMillis() - startTime);
                 }
                 // pass the results into the view.
-                model.put("results" + i, results);
+				QuickstartResults quickstartResults = new QuickstartResults(
```
Comment (willwarren) [inline src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java]: Can you set your tab size to 4 spaces.

File: src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java [hunk 2]
```diff
@@ -251,8 +263,23 @@ protected ModelAndView handleSearch(HttpServletRequest request, HttpServletRespo
...
+			String strategy = "[{ 'terms': 2, 'mustMatch': 2 }, { 'terms': 3, 'mustMatch': 2 }, { 'terms': 4, 'mustMatch': 3 }, { 'terms': 5, 'mustMatch': 3 }, { 'terms': 6, 'mustMatch': 4 }, { 'terms': 7, 'mustMatch': 4 }, { 'terms': 8, 'mustMatch': 5 }, { 'termsGreaterThan': 8, 'mustMatch': 60, 'percentage': true }]";
+			try {
+				strategy = matchStrategies[i].trim();
+				MatchStrategy oMatchStrategy = Utils.getMatchStrategy(strategy);
+				if (oMatchStrategy != null) {
+					query.setMatchStrategy(oMatchStrategy);
+				}
+
+			} catch (Exception e) {
+				// swallow any exception in attempting to get / set a match
```
Comment (willwarren) [inline src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java]: Never swallow exceptions.  `e.printStackTrace()`

File: src/main/java/com/groupbyinc/quickstart/helper/Utils.java [hunk 3]
```diff
@@ -30,4 +39,37 @@
...
+		if (!(jsonString.startsWith("[") && jsonString.endsWith("]"))) {
+			jsonString = "[" + jsonString + "]";
+		} else if (!jsonString.startsWith("[") || !jsonString.endsWith("]")) {
+			return null;
+		}
+		ObjectMapper mapper = new ObjectMapper();
+		List<PartialMatchRule> rules = null;
+		try {
+			rules = mapper.readValue(jsonString, new TypeReference<List<PartialMatchRule>>(){});
+		} catch (JsonParseException e) {
```
Comment (willwarren) [inline src/main/java/com/groupbyinc/quickstart/helper/Utils.java]: This should all be one exception.

File: src/main/java/com/groupbyinc/quickstart/model/QuickstartResults.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package com.groupbyinc.quickstart.model;
```
Comment (willwarren) [inline src/main/java/com/groupbyinc/quickstart/model/QuickstartResults.java]: Could you create a ticket marked as technical debt to add matchStrategy to the actual results object?

**Comment by JerryCB** [comment]
@willwarren Your code review requests have been implemented, and the code committed.

**Comment by codeallthethingz** [comment]
work merged into docker branch and now merged to develop main.

**Pull request closed without merging**